### PR TITLE
Refactor Python dep handling + update cheat-sheet

### PR DESCRIPTION
**Pull request by AltGr**
This started with just an update of the cheat-sheet, but since I encountered LaTeX compilation issues due to Pygments, ended up including a refactor of the Python dependencies handling in our builds.

Please let me know if I should split the two in separate PRs (e.g. if some more discussion on the Python end is needed)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,10 +32,35 @@ dependencies-ocaml-with-z3:
...
+	test -d $(PY_VENV_DIR) || python3 -m venv $(PY_VENV_DIR)
+	$(PY_VENV_ACTIVATE) python3 -m pip install -U \
+	  -e runtimes/python/catala \
+	  -e syntax_highlighting/en/pygments \
+	  -e syntax_highlighting/fr/pygments \
+	  -e syntax_highlighting/pl/pygments
+	touch $@
+
+# Run sub-make within the Python venv
+MAKEP = $(PY_VENV_ACTIVATE) $(MAKE)
```
Comment (denismerigoux) [inline Makefile:56]: Yeah that's a pain. We really need to tell in the `README` to people to use the `Makefiles` to get literate outputs.

File: syntax_highlighting/fr/catala_fr.iro [hunk 2]
```diff
@@ -216,7 +216,7 @@ code : context {
   }
 
   : pattern {
-    regex \= \b(selon|sous\s+forme|fixé|par|décroissante|croissante|varie|avec|on\s+a|soit|dans|champ\s+d'application|dépend\s+de|déclaration|inclusion|contenu|règle|sous\s+condition|condition|donnée|conséquence|rempli|égale\s+à|assertion|définition|état|étiquette|exception)\b
+    regex \= \b(selon|sous\s+forme|fixé|par|décroissante|croissante|varie|avec|on\s+a|soit|dans|champ\s+d'application|dépend\s+de|déclaration|inclusion|contenu|règle|sous\s+condition|condition|donnée|conséquence|rempli|égal\s+à|assertion|définition|état|étiquette|exception)\b
```
Comment (denismerigoux) [inline syntax_highlighting/fr/catala_fr.iro:219]: YES this was broken for me.

Comment (denismerigoux) [reply syntax_highlighting/fr/catala_fr.iro:219]: Could you check whether all new keywords are in there? For instance `résultat` is missing for me.

**Review by denismerigoux — APPROVED** [review]
Thanks! I approve the changes. They'll break people's workflow when pulling from `master` so I suggest you make a little annoucement on the Zulip to everybody when this is merged.

File: syntax_highlighting/fr/catala_fr.iro (hunk 2 above)
Comment (AltGr) [reply syntax_highlighting/fr/catala_fr.iro:219]: You need to rebuild the pygments lexer (`make pygmentize_fr`), all should be there
(at least all that is in the cheat-sheet, that's how I checked :p )

**Comment by AltGr** [comment]
Ah this will also include doc + CI builds within the artifacts newly published to https://catalalang.github.io/catala once it's merged :crossed_fingers: 

> I suggest you make a little annoucement on the Zulip to everybody when this is merged.

Ok, I plan to do both these things tomorrow then!

**Pull request merged**